---
title: "BlackGillShrimpDADA2"
author: "Jennifer Ouyang"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
#source("https://raw.githubusercontent.com/joey711/phyloseq/master/inst/scripts/installer.R",
       local = TRUE)
#install_phyloseq(branch = "devel")
```

```{r}
library(readr)
library(fs)
library(R.utils)
library(tidyverse)
library(stringr)
library(phyloseq)
library(ggplot2)
library(magrittr)
library(dplyr)

```


```{r}
library(dada2); packageVersion("dada2")

path <- "C:/Users/Jennifer Ouyang/RStudio/MarineLab/src"
list.files(path)

# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="R2_001.fastq", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```

##Inspect read quality profiles

```{r}
plotQualityProfile(fnFs[1:2])
```

```{r}
plotQualityProfile(fnRs[1:2])
```

##Filter and trim

```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```


```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,230),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE
head(out)
```



##Learn the Error Rates
```{r}
errF <- learnErrors(filtFs, multithread=TRUE)
```

```{r}
errR <- learnErrors(filtRs, multithread=TRUE)
```

```{r}
plotErrors(errF, nominalQ=TRUE)
```


##Sample Inference
```{r}
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
dadaFs[[1]]
```

##Merge paired reads
```{r}
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
```

##Construct sequence table
```{r}
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```


```{r}
# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
```

##Remove chimeras
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
```

```{r}
sum(seqtab.nochim)/sum(seqtab)
```


##Track reads through the pipeline
```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)

```

##Assign taxonomy
```{r}
taxa <- assignTaxonomy(seqtab.nochim, "C:/Users/Jennifer Ouyang/RStudio/MarineLab/src/silva_nr99_v138.1_train_set.fa.gz", multithread=TRUE)
```


```{r}
taxa.print <- taxa # Removing sequence rownames for display only
rownames(taxa.print) <- NULL
head(taxa.print)
```


#Phyloseq
##Map Data
##Make a Phyloseq Object
```{r}
#Read files for phyloseq
R1.fileName <- list.files(path, pattern="R1_001.fastq", full.names = TRUE)
R2.fileName <- str_replace(R1.fileName,"R1","R2")
sampleInfo <- data.frame(R1.fileName, R2.fileName)
sampleInfo %<>%
  separate(R1.fileName, into = "sampleName", sep = "_", remove=FALSE)

```

```{r}
otus = otu_table(seqtab.nochim, taxa_are_rows=FALSE)

sampleInfo %>%
  select(sampleName,R1.fileName) %>%
  column_to_rownames(var = "sampleName") ->
  sd

ps <- phyloseq(otus,
               sample_data(sd),
               tax_table(taxa))
```

```{r}
print(ps)
```

## Save Phyloseq to RDS
```{r}
write_rds(ps, ps.file)
```

## Check Phyloseq RDS
```{r}
print(read_rds(ps.file))
```


```{r}
sessionInfo()
```